frontend/ui/widget/touchmenu.lua

```diff
@@ -744,6 +746,105 @@ function TouchMenu:updateItems()
     end)
 end
 
+local MAX_MENU_DEPTH = 20 -- currently our menu needs at least 12 here
+function TouchMenu:_recurse(val, path, text, search_for, depth)
```

Comment:
This could get the `results` array as a first argument, where results would be accumulated - instead of `table.insert(TouchMenu.foundMenuItem)`